sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/aio/_client.py

```diff
@@ -324,6 +330,7 @@ class KeyClient(AsyncKeyVaultClientBase):
         )
         return Key._from_key_bundle(bundle)
 
+    @distributed_trace
     def list_keys(self, **kwargs: Mapping[str, Any]) -> AsyncIterable[KeyBase]:
```

Comment:
Also have a look at async list traces, they're an interesting case. This method isn't async, and quickly returns a lazy iterable without using the network. When the iterable is iterated at some unknown future time, it will make async network requests.

Reply:
![image](https://user-images.githubusercontent.com/13604256/61565951-771bb480-aa2f-11e9-9c77-5337973c28e3.png)
 is generated from 
```
import time
from azure.core.settings import settings
from azure.keyvault.keys.aio import KeyClient
from azure.identity.aio import DefaultAzureCredential
from opencensus.ext.zipkin.trace_exporter import ZipkinExporter
from opencensus.trace.samplers import AlwaysOnSampler
from opencensus.trace import tracer as tracer_module
import asyncio

# MAKE SURE TO SET THE APPROPRIATE CREDNTIAL ENVIRONMENT VARIABLES
azure_credential = DefaultAzureCredential()

settings.tracing_implementation.set_value("opencensus")
zipExportInst = ZipkinExporter(service_name="suyog-azure-core-v0.01-opencensus", host_name="localhost", port=9411)
tracer = tracer_module.Tracer(sampler=AlwaysOnSampler(), exporter=zipExportInst)

async def run():
    with tracer.span("parent") as parent:
        client = KeyClient("https://susoti-vault.vault.azure.net/", azure_credential)

        for index in range(2):
            await client.create_key("key-{}".format(index), "RSA")

        keys = client.list_keys(max_page_size=10)

        with tracer.span("first page"):
            async for key in keys:
                print("{} -> {}".format(key.name, key.id))
                break

        key = await client.get_key("key-name")
        print("{} -> {}".format(key.name, key.id))

        with tracer.span("do something else"):
            time.sleep(0.1)

        with tracer.span("Rest of the pages") as chld:
            async for key in keys:
                print("{} -> {}".format(key.name, key.id))

tracer.finish()

asyncio.run(run())
```